misk/src/main/kotlin/misk/client/ClientInvocationHandler.kt

```diff
@@ -117,84 +106,11 @@ internal class ClientInvocationHandler(
   override fun invoke(proxy: Any, method: Method, args: Array<out Any>?): Any {
     val argsList = args?.toList() ?: listOf()
 
-    val action = actionsByMethod[method.name] ?: throw IllegalStateException(
-        "no action corresponding to ${interfaceType.qualifiedName}#${method.name}"
-    )
-
     val retrofitProxy = proxiesByMethod[method.name] ?: throw IllegalStateException(
         "no action corresponding to ${interfaceType.qualifiedName}#${method.name}"
     )
-    val interceptors = (interceptorsByMethod[method.name] ?: listOf())
-
-    val beginCallInterceptors = interceptors + RetrofitCallInterceptor(retrofitProxy, method)
-    val beginCallChain = RealBeginClientCallChain(action, argsList, beginCallInterceptors)
-    val wrappedCall = beginCallChain.proceed(beginCallChain.args)
-
-    val requestInterceptors = interceptors + RetrofitRequestInterceptor(wrappedCall)
-    return InterceptedCall(action, requestInterceptors, argsList, wrappedCall)
-  }
-
-  /** Wraps a retrofit [Call] to invoke interceptors before handing off to Retrofit */
-  private class InterceptedCall(
-    private val action: ClientAction,
-    private val interceptors: List<ClientApplicationInterceptor>,
-    private val args: List<*>,
-    private val wrapped: Call<Any>
-  ) : Call<Any> {
-    override fun enqueue(callback: Callback<Any>) {
-      val allInterceptors = interceptors + RetrofitRequestInterceptor(wrapped)
-      RealClientChain(action, args, wrapped, callback, allInterceptors).proceed(args, callback)
-    }
-
-    override fun execute(): Response<Any> {
```

Comment:
# ⚠️ Behavior change!

Previously this abstraction caused all network calls to use a thread from OkHttp’s dispatcher thread pool. With this change synchronous calls won’t use that extra thread.